source/simulator/core.cc

```diff
@@ -1201,6 +1199,13 @@ namespace aspect
       pcout.get_stream().imbue(s);
     }
 
+    //We need to setup the free surface degrees of freedom first if there is a free
+    //surface active, since the mapping must be in place before applying boundary
+    //conditions that rely on it (such as no flux bcs).
+    if (parameters.free_surface_enabled)
+      free_surface->setup_dofs();
```

Comment:
You may even be able to do this in the constructor of `FreeSurfaceHandler` already, or does it rely on something that's not yet available at that time?